docs/build.md

```diff
@@ -1,8 +1,126 @@
-* Build custom software
-
-The build code keeps a cache of all packages source code (Git/Mercurial/SVN)
-repositories so that they do not need to be downloaded on every build. If it is
-desired to change the source code of a package, you may edit the package local
-repository content. To build the package from this modified local repository,
-either use a file:// URL pointing to it in the package metadata (YAML)
-file or disable package repositories update in config/host_os.yaml.
+# Building process
+
+The build process relies on a metadata
+[repository](https://github.com/open-power-host-os/versions) to obtain
+information about the components to build, such as URLs, versions,
+commit IDs, dependencies and distro-specific build parameters.
+
+During the build process, the build script checks out the metadata
+(versions) repository, obtains the information necessary to build the
+package and starts an instance of
+[mock](https://github.com/rpm-software-management/mock/wiki) that
+builds the package in a chroot, taking care of its build dependencies.
+
+Other functionalities of the build script are:
```

Comment:
I get your point but I want to give an overview of what the script does so that the user doesn't get completely confused when `./host_os.py --help` shows a bunch of weird stuff. 

I don't think the other commands are useful to anyone besides us (perhaps `update-versions`?) and, as you know, IMO they shouldn't even be in the same repository. So I think having a "release process" documentation makes no sense if nobody cares about it.

But I'm open to discussion. Maybe we can find a middle ground.

*As a side note:
I don't even believe in the "release" concept anymore, maybe we could just keep it rolling, drop `update-versions` and `update-metapackage` and tweak `build-release-notes` to be a bit more process agnostic.*